internal/packer/packer.go

```diff
@@ -1757,6 +1762,67 @@ func (pack *Pack) logic(programName string) error {
 	return nil
 }
 
+// kernelGoarch returns the GOARCH value that corresponds to the provided
+// vmlinuz header. It returns one of "arm64", "arm", "amd64", or the empty
+// string if not detected.
+func kernelGoarch(hdr []byte) string {
+	const (
```

Comment:
Can you add a comment to explain what these magic values actually are? Does the Linux kernel define these somewhere we could link to? Or are they not arbitrary values from the kernel at all?